ecommerce/app/controllers/items_controller.rb

```diff
@@ -0,0 +1,66 @@
+class ItemsController < ApplicationController
+  before_action :set_item, only: [:show, :edit, :update, :destroy]
+
+  # GET /items
+  # GET /items.json
+  def index
+    @items = Item.all
+  end
+
+  # GET /items/new
+  def new
+    @item = Item.new
+  end
+  # POST /items
+  # POST /items.json
+  def create
+    @item = Item.new(item_params)
+
+      respond_to do |format|
+        if @item.save
+          format.html { redirect_to @item, notice: 'Item was successfully created.' }
+          format.json { render :show, status: :created, location: @item }
+        else
+          format.html { render :new }
+          format.json { render json: @item.errors, status: :unprocessable_entity }
+        end
+      end
+    end
+
+  # PATCH/PUT /items/1
+  # PATCH/PUT /items/1.json
+  def update
+      respond_to do |format|
+        if @item.update(item_params)
+          format.html { redirect_to @item, notice: 'Item was successfully updated.' }
+          format.json { render :show, status: :ok, location: @item }
+        else
+          format.html { render :edit }
+          format.json { render json: @item.errors, status: :unprocessable_entity }
+        end
+      end
+    end
+
+  # DELETE /items/1
+  # DELETE /items/1.json
+  def destroy
+      set_item
+      Order.where(item_name: @item.name.to_s).destroy_all
```

Comment:
never use string while validating update or delete operations,
Reduce lines 47, 48 lines to single line